hooks/command

```diff
@@ -13,60 +13,121 @@ fi
 ((timeout=BUILDKITE_TIMEOUT*60))
 export BUILDKITE_TIMEOUT
 
+# Default values can be overridden by setting "K8S_PLUGIN_*" env vars as used below.
+readonly job_apply_loop_interval_seconds="${K8S_PLUGIN_JOB_APPLY_LOOP_INTERVAL:-5}"
+readonly job_apply_loop_timeout_seconds="${K8S_PLUGIN_JOB_APPLY_LOOP_TIMEOUT:-120}"
+readonly log_loop_interval_seconds="${K8S_PLUGIN_LOG_LOOP_INTERVAL:-3}"
+readonly log_loop_timeout_seconds="${K8S_PLUGIN_LOG_LOOP_TIMEOUT:-5}"
+readonly log_loop_snapshot_limit_bytes="${K8S_PLUGIN_LOG_LOOP_SNAPSHOT_LIMIT_BYTES:-1024}"
+readonly job_status_loop_sleep_interval="${K8S_PLUGIN_JOB_STATUS_LOOP_INTERVAL:-5}"
+readonly log_complete_loop_interval_seconds="${K8S_PLUGIN_LOG_COMPLETE_LOOP_INTERVAL:-1}"
+readonly log_complete_loop_timeout_seconds="${K8S_PLUGIN_LOG_COMPLETE_LOOP_TIMEOUT:-30}"
+readonly use_agent_node_affinity=${BUILDKITE_PLUGIN_K8S_USE_AGENT_NODE_AFFINITY:-false}
+readonly print_resulting_job_spec=${BUILDKITE_PLUGIN_K8S_PRINT_RESULTING_JOB_SPEC:-false}
```

Comment:
Do you want to add yaml versions of these environment variables or just document how to set them and what they mean?

I'm a bit hesitant about yaml, but mostly because I'm lazy and I think most people will never change defaults 😊

Reply:
There were two cases I had in mind, this particular one refers only to this variable `BUILDKITE_PLUGIN_K8S_PRINT_RESULTING_JOB_SPEC` which would be nice have as a regular config - this I'd prefer to have both documented in the README and added to the yaml.

[The other case](https://github.com/EmbarkStudios/k8s-buildkite-plugin/pull/14#discussion_r469362676) refers to all the other new variables since they're quite lower level, implementation specific and as you mentioned, the defaults are even less likely to be changed by most users. These could only be documented in the README so that at least people know that they exist if they need/would like to change them. They won't be in the yaml file since it's not supported that they're configurable via the plugin's section in a pipeline config, rather they'll be directly passed onto the agent.

Reply:
Got it, I'll move `BUILDKITE_PLUGIN_K8S_PRINT_RESULTING_JOB_SPEC` to YAML, if something else catches my eye, I'll do that too